public/js/invertedIndex.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * The Inverted Index class.
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.unique = [];
+    this.tableObj = {};
+  }
+
+  /**
+   * returns the object containing all the uploadef files
+   * @return {Object} Index of all the files
+   */
+  searchAllFiles() {
+    return this.tableObj;
+  }
+
+  /**
+   * A method that tokenizes the string that is passed through it
+   * @param  {string} data the string that is gotten from the "text" key in the JSON object
+   * @return {Array} containing String.
+   */
+  tokenize(data) {
+    this.terms = data.replace(/[^\w\s]/gi, ' ')
+      .match(/\w+/g);
+    return this.terms;
+  }
+
+  /**
+   * A method that filters the array passed into it for unique words
+   * @param {String} data Array of strings
+   * @returns {String} Returns array of unique words
+   */
+  uniqueWords(data) {
+    this.unique = [...new Set(data)];
```

Comment:
Thank you, i am currently using the `this.unique` variable elsewhere in the code, hence my need to store it before I return it.